src/runtime.cc

```diff
@@ -12,6 +12,7 @@
 
 #include <mpi.h>
 
+#include "affinity.imp.h"
```

Comment:
The `imp` implies that this is an implementation detail, which means it shouldn't be part of the public interface and therefore this should not be how the affinity header is included. This should just be `#include "affinity.h"` and the implementation details should be handled inside this header.

I would get rid of the `affinity.imp.h` entirely, and just directly include the platform specific headers inside `affinity.h`.